packages/inferno-from-react-generator/src/expressions/inferno-hooks-variable-statement-wrapper.ts

```diff
@@ -21,13 +23,21 @@ export class InfernoHooksVariableStatementWrapper extends VariableStatement {
     super([], declarationList);
     this.hasExport = modifiers.indexOf(SyntaxKind.ExportKeyword) !== -1;
     this.componentInfo = componentInfo;
-    componentDeclaration.name = new Identifier(`React${componentDeclaration.name.toString()}`);
-    componentDeclaration.initializer = (componentDeclaration.initializer as Call)
-      .argumentsArray[0];
+    if (componentDeclaration.initializer instanceof Call) {
+      const call = componentDeclaration.initializer;
+      if (call.expression.toString() === 'forwardRef') {
+        componentDeclaration.name = new Identifier(`React${componentDeclaration.name.toString()}`);
+      }
+      if (call.expression.toString() === 'memo' || call.expression.toString() === 'React.memo') {
+        this.pureComponent = true;
+      }
```

Comment:
Let's not call toString three times here:

```suggestion
      const call = componentDeclaration.initializer;
      const expressionStr = call.expression.toString();
      if (expressionStr === 'forwardRef') {
        componentDeclaration.name = new Identifier(`React${componentDeclaration.name.toString()}`);
      }
      if (expressionStr === 'memo' || expressionStr === 'React.memo') {
        this.pureComponent = true;
      }
```